app/controllers/v3/service_credential_bindings_controller.rb

```diff
@@ -120,13 +120,16 @@ def parameters
   private
 
   def create_key_binding(message, service_instance)
-    V3::ServiceCredentialBindingKeyCreate.new.precursor(
-      service_instance,
-      message.name
-    )
+    action = V3::ServiceCredentialBindingKeyCreate.new(user_audit_info, message.audit_hash)
+    binding = action.precursor(service_instance, message.name)
 
-    head :not_implemented
-    return
+    case service_instance
+    when ManagedServiceInstance
+      pollable_job_guid = enqueue_key_job(binding.guid, message)
+      head :accepted, 'Location' => url_builder.build_url(path: "/v3/jobs/#{pollable_job_guid}")
+    else
+      head :not_implemented
+    end
```

Comment:
Keys cannot be created for UPSI (the precursor will return error on validation) so there is no need for the switch here

Reply:
Thanks for spotting this!